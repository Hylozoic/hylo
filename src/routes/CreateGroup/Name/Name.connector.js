import { connect } from 'react-redux'
import { push } from 'connected-react-router'
import { addGroupName, addParentIds } from '../CreateGroup.store'
import getRouteParam from 'store/selectors/getRouteParam'
import { get } from 'lodash/fp'

export function mapStateToProps (state, props) {
  const parentId = getRouteParam('groupId', state, props)
  return {
    parentIds: [parentId],
    groupName: get('name', state.CreateGroup)
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    goToNextStep: () => dispatch(push('/create-group/domain')),
    goHome: () => dispatch(push('/')),
    addGroupName: name => dispatch(addGroupName(name)),
    addParentIds: parentIds => dispatch(addParentIds(parentIds))
  }
}

export function mergeProps (stateProps, dispatchProps, ownProps) {
  const { parentId } = stateProps
  const addParentIds = parentId
    ? () => dispatchProps.addParentIds([parentId])
    : () => {}

  return {
    ...stateProps,
    ...dispatchProps,
    ...ownProps,
    addParentIds
  }
}

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)
