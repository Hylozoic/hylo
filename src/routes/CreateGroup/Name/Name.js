import React, { Component } from 'react'
import '../CreateGroup.scss'
import TextInput from 'components/TextInput'
import { hyloNameWhiteBackground, happyAxolotl } from 'util/assets'
import { bgImageStyle } from 'util/index'
import ModalFooter from 'components/ModalFooter'
import ModalSidebar from 'components/ModalSidebar'

export default class Name extends Component {
  constructor (props) {
    super(props)
    this.state = {
      groupName: ''
    }
  }

  handleNameChange = (event) => {
    const groupName = event.target.value
    this.setState({
      groupName
    })
  }

  submit = () => {
    this.props.addGroupName(this.state.groupName)
    this.props.goToNextStep()
  }

  errorCheckAndSubmit = () => {
    if (this.state.groupName === '') {
      this.setState({
        error: 'Please enter a group name'
      })
    } else {
      this.submit()
    }
  }

  onEnter = event => {
    if (event.key === 'Enter') {
      this.errorCheckAndSubmit()
    }
  }

  componentWillMount = () => {
    const { groupName } = this.props
    if (groupName) this.setState({ groupName })
  }

  componentDidMount = () => {
    this.props.addParentIds()
  }

  render () {
    return <div styleName='flex-wrapper'>
      <ModalSidebar
        imageDialogOne={"Hi, I'm Hylo the Axolotl!"}
        imageDialogTwo={"It's great to meet you!"}
        imageUrl={happyAxolotl}
        onClick={this.props.goHome}
        theme={sidebarTheme}
        header="Great, let's get started"
        body="All good things start somewhere! Let's kick things off with a catchy name for your group."
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 1/3</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-vertically'>
          <span styleName='text-input-label'>{ this.state.groupName && "What's the name of your group?"}</span>
          <TextInput
            type='text'
            name='group-name'
            onChange={this.handleNameChange}
            value={this.state.groupName}
            theme={inputTheme}
            placeholder="What's the name of your group?"
            noClearButton
            onEnter={this.onEnter}
          />
          { this.state.error && <span styleName='arrow-up' /> }
          { this.state.error && <span styleName='error'>{this.state.error}</span>}
        </div>
      </div>
      <ModalFooter
        submit={this.errorCheckAndSubmit}
        showPrevious={false}
        continueText={'Continue'}
      />
    </div>
  }
}

const inputTheme = {
  inputStyle: 'modal-input',
  wrapperStyle: 'center'
}

const sidebarTheme = {
  sidebarHeader: 'sidebar-header-full-page',
  sidebarText: 'gray-text sidebar-text-full-page'
}
