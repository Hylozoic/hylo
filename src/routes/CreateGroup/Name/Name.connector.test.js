import { mapDispatchToProps, mapStateToProps } from './Name.connector'

const dispatch = jest.fn(x => x)
const props = { match: { params: { } } }
const dispatchProps = mapDispatchToProps(dispatch, props)

describe('Domain', () => {
  it('should call fetchGroup from mapDispatchToProps', () => {
    const groupName = 'Group Name'
    expect(dispatchProps.addGroupName(groupName)).toMatchSnapshot()
  })

  it('should call goToNextStep from mapDispatchToProps', () => {
    expect(dispatchProps.goToNextStep()).toMatchSnapshot()
  })

  it('should call goHome from mapDispatchToProps', () => {
    expect(dispatchProps.goHome()).toMatchSnapshot()
  })

  it('should have groupName in mapStateToProps', () => {
    const groupName = 'groupName'
    const state = {
      CreateGroup: {
        name: groupName
      }
    }
    expect(mapStateToProps(state, props).groupName).toBe(groupName)
  })
})
