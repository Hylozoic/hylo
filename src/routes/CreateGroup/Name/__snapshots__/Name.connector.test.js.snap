// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Domain should call fetchGroup from mapDispatchToProps 1`] = `
Object {
  "payload": "Group Name",
  "type": "CreateGroup/ADD_GROUP_NAME",
}
`;

exports[`Domain should call goHome from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Domain should call goToNextStep from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/create-group/domain",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;
